app/models/world_location.rb

```diff
@@ -67,14 +67,50 @@ def self.travel_rules
   end
 
   def covid_status
+    current_covid_status_data&.dig("covid_status")
+  end
+
+  def next_covid_status
+    next_covid_status_data&.dig("covid_status")
+  end
+
+  def next_covid_status_applies_at
+    Time.zone.parse(next_covid_status_data["covid_status_applies_at"])
+  rescue NoMethodError, TypeError
+    nil
+  end
+
+  def current_covid_status_data
+    current_statuses = covid_status_data_for_location&.select do |status|
+      start_date = Time.zone.parse(status["covid_status_applies_at"])
+      start_date.past?
+    end
+
+    return if current_statuses.blank?
+
+    current_statuses.max_by { |status| status["covid_status_applies_at"] }
+  end
+
+  def next_covid_status_data
```

Comment:
These all look good. I'm wondering if we should move the initialiser up to above the instance methods to be more conventional?

Reply:
Oops! Will move.